packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/Expressions/KnownValueExpressionsTests.cs

```diff
@@ -12,6 +12,20 @@ namespace Microsoft.Generator.CSharp.Tests.Expressions
 {
     public class KnownValueExpressionsTests
     {
+        [Test]
+        public void NullConditionalOperatorNotChangeOriginalType()
+        {
+            VariableExpression original = new VariableExpression(typeof(int), "foo");
+
+            using CodeWriter writer = new();
+            var nullableExpression = original.As<int>().NullConditional(true);
+            nullableExpression.Write(writer);
```

Comment:
how about we write another expression after the null conditional to verify that we could chain the call as well?
For instance, we change the type `int` to `BinaryData`,
```csharp
var original = new VariableExpression(typeof(BinaryData), "foo");

var nullableExpression = original.As<BinaryData>().NullConditional(true);
var result = nullableExpression.ToMemory();
result.Write(writer);
```